src/test-utils/index.tsx

```diff
@@ -179,24 +183,64 @@ export const openStoryPage = ({
         isDarkMode,
     });
 
+const hydrateSSRPage = async (page: PageApi): Promise<void> => {
+    await page.evaluate(() => {
+        // @ts-expect-error this executes in the browser, see ssr.tsx
+        window.hydrate();
+    });
+};
+
+const checkHydrationMissmatch = async (page: PageApi): Promise<void> => {
+    const {testPath, currentTestName} = expect.getState();
+    const tmpdir = os.tmpdir();
+    const snapshotId = kebabCase(`${path.basename(testPath)}-${currentTestName}`);
+    const baselineImagePath = path.join(tmpdir, `${snapshotId}-snap.png`);
+    await page.screenshot({path: baselineImagePath});
+
+    await hydrateSSRPage(page);
+    const hydrateImage = await page.screenshot();
+
+    try {
+        expect(hydrateImage).toMatchImageSnapshot({
+            // use the saved ssr snapshot as the baseline
+            customSnapshotsDir: tmpdir,
+            customSnapshotIdentifier: snapshotId,
+            customDiffDir: path.join(
+                __dirname,
+                '..',
+                '__acceptance_tests__',
+                '__image_snapshots__',
+                '__diff_output__'
+            ),
+        });
+    } catch (error: any) {
+        Error.captureStackTrace(error, checkHydrationMissmatch);
+        throw error;
+    } finally {
+        fs.unlinkSync(baselineImagePath);
+    }
+};
+
 /**
  * Renders a page with a React component in the server and opens it in the browser, where it's hydrated client side.
  * `name` is the name (without extension) of a file in the __ssr_pages__ folder. This file exports the component to be rendered.
  */
-export const openSSRPage = ({
+export const openSSRPage = async ({
     name,
     device = TABLET_DEVICE,
     skin = MOVISTAR_SKIN,
+    checkHidrationVisualMissmatch = true,
```

Comment:
I've enabled this new check in all the SSR tests by default. But I have disabled it in some of them